geomesa-fs/geomesa-fs-storage/geomesa-fs-storage-parquet/src/main/scala/org/locationtech/geomesa/fs/storage/parquet/io/SimpleFeatureWriteSupport.scala

```diff
@@ -96,15 +118,13 @@ object SimpleFeatureWriteSupport {
 
   // TODO support z/m
   private def geometry(name: String, index: Int, binding: ObjectType): AttributeWriter[_] = {
-    binding match {
-      case ObjectType.POINT           => new PointAttributeWriter(name, index)
-      case ObjectType.LINESTRING      => new LineStringAttributeWriter(name, index)
-      case ObjectType.POLYGON         => new PolygonAttributeWriter(name, index)
-      case ObjectType.MULTIPOINT      => new MultiPointAttributeWriter(name, index)
-      case ObjectType.MULTILINESTRING => new MultiLineStringAttributeWriter(name, index)
-      case ObjectType.MULTIPOLYGON    => new MultiPolygonAttributeWriter(name, index)
-      case ObjectType.GEOMETRY        => new GeometryWkbAttributeWriter(name, index)
-      case _ => throw new IllegalArgumentException(s"Can't serialize field '$name' of type $binding")
+    val geomTypes = Seq(ObjectType.POINT, ObjectType.LINESTRING, ObjectType.POLYGON, ObjectType.MULTIPOINT,
+      ObjectType.MULTILINESTRING, ObjectType.MULTIPOLYGON, ObjectType.GEOMETRY)
+
+    if (geomTypes.contains(binding)) {
```

Comment:
i think this is another case where we won't ever get here unless this check is true, so we can skip it.